querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java

```diff
@@ -82,4 +82,15 @@ public FactoryExpressionBase(Class<? extends T> type) {
         return new FactoryExpressionWrapper<T>(this);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        } else if (o instanceof FactoryExpression) {
+            return getClass().equals(o.getClass()) && getArgs().equals(((FactoryExpression) o).getArgs());
```

Comment:
Ok. I'll continue working in the evening.
On 31 May 2015 16:11, "Ruben Dijkstra" <notifications@github.com> wrote:

> In
> querydsl-core/src/main/java/com/querydsl/core/types/FactoryExpressionBase.java
> <https://github.com/querydsl/querydsl/pull/1386#discussion_r31390565>:
>
> > @@ -82,4 +82,15 @@ public FactoryExpressionBase(Class<? extends T> type) {
> >          return new FactoryExpressionWrapper<T>(this);
> >      }
> >
> > +    @Override
> > +    public boolean equals(Object o) {
> > +        if (o == this) {
> > +            return true;
> > +        } else if (o instanceof FactoryExpression) {
> > +            return getClass().equals(o.getClass()) && getArgs().equals(((FactoryExpression) o).getArgs());
>
> Shall I do that then?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/querydsl/querydsl/pull/1386/files#r31390565>.
>